pkg/pillar/types/global.go

```
@@ -114,6 +117,230 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+	log.Debugf("Adding bool item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:    CONFIGITEM_TYPE_BOOL,
+		Key:         key,
+		boolDefault: defaultBool,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	} else {
+		configMap.itemMap[key] = configItem
+	}
+	log.Debugf("Added bool item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddStringItem(key string, defaultString string, agentSetting bool) {
+	log.Debugf("Adding string item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:      CONFIGITEM_TYPE_STRING,
+		Key:           key,
+		stringDefault: defaultString,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	} else {
+		configMap.itemMap[key] = configItem
+	}
+	log.Debugf("Added string item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddTriStateItem(key string, defaultTriState TriState, agentSetting bool) {
+	log.Debugf("Adding tristate item %s", key)
+	configItem := ConfigItemSpec{
+		Key:             key,
+		ItemType:        CONFIGITEM_TYPE_TRISTATE,
+		triStateDefault: defaultTriState,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	} else {
+		configMap.itemMap[key] = configItem
+	}
+	log.Debugf("Added tristate item %s", key)
+}
+
+type ConfigItem struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intValue	    uint32
+	stringValue		string
+	boolValue	    bool
+	triStateValue	TriState
+}
+
+type ConfigItemMap struct {
+	globalSettings map[string]ConfigItem
+	agentSettings  map[string]ConfigItem
+}
+
+func (specMap *ConfigItemSpecMap) parseItemTestVersion(configMap *ConfigItemMap, item config.ConfigItem) {
+	key := item.Key
+	configItem := ConfigItem {
+		Key:	key,
+	}
+	if specMap.itemMap[key].ItemType == CONFIGITEM_TYPE_INT {
+		i64, err := strconv.ParseUint(item.Value, 10, 32)
+		if err == nil {
+			configItem.intValue = uint32(i64)
+		} else {
+			configItem.intValue = specMap.itemMap[key].intDefault
+			log.Errorf("parseConfigItems: %s", err)
```

Add a proper message here. When the user sees the message,he should immediately understand what mistake he did. And it is not just logging the error - You should return the error back too.